github: fix codeowners

whilei opened the pull request:
Rel #34

These are not the _eternal_ or canonical
codeowners (I hope), but can serve for now
as a starting point.

They should continue to be revised as
discussions around this topic progress.

whilei committed:
github: fix codeowners

Rel #34

These are not the _eternal_ or canonical
codeowners (I hope), but can serve for now
as a starting point.

They should continue to be revised as
discussions around this topic progress.

sorpaas reviewed (changes requested):
Ah @whilei, I realize, that not all folders have code owners. The original purpose is probably just to notify those people if the files under those folders are changed (because they're the most knowledgeable). So I'm thinking, what do you think about if we simply remove `CODEOWNERS` file?

whilei committed:
Remove CODEOWNERS

Not all folders are assigned owners, and
with a small set of active contributors the
tooling this provides is too limited to
warrant maintenance.
